src/BSON/UTCDateTime.stub.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/** @generate-function-entries */
+
+namespace MongoDB\BSON
+{
+    final class UTCDateTime
+    {
+#if PHP_VERSION_ID >= 80000
+        public final function __construct(int|string|float|\DateTime|\DateTimeImmutable|null $milliseconds = null) {}
```

Comment:
I suppose [DateTimeInterface](https://www.php.net/manual/en/class.datetimeinterface.php) isn't feasible to use, since we actually depend on the specific DateTime or DateTimeImmutable classees to access internal fields and don't merely invoke interface methods (i.e. the construct would fail to handle a userland class implementing DateTimeInterface, assuming that's even possible).

Reply:
The check currently in the constructor currently only accepts `DateTime` or `DateTimeImmutable` instances, but that could be changed. However, we then access the object through `Z_PHPDATE_P`, which in turn calls [`php_date_obj_from_obj`](https://heap.space/xref/PHP-8.1/ext/date/php_date.h?r=c85fcbc2#46). As `DateTimeInterface` technically can't be implemented by user land code and was specifically created to accept both `DateTime` and `DateTimeImmutable` instances (see [documentation](https://www.php.net/manual/en/class.datetimeinterface.php)) we can also try changing it if you think it's safe to do so.

Reply:
Ah, I forgot that `DateTimeInterface` was only available for internal use. We actually use that in our documentation for [`UTCDateTime::__construct()`](https://www.php.net/manual/en/mongodb-bson-utcdatetime.construct.php) so I think it'd make sense to do so here, too. See: https://github.com/mongodb/mongo-php-driver/blob/1.14.0/src/BSON/UTCDateTime.c#L152-L161

Reply:
Changed as suggested 👍